File: Minimum_Replacements_to_Sort_the_Array.py
class Solution:
    def minimumReplacement(self, nums: List[int]) -> int:
        current_largest = nums[-1]
        total_replacements = 0
        
        for i in range(len(nums) - 1, -1, -1):
            if nums[i] <= current_largest:
                current_largest = nums[i]
                continue

            if nums[i] % current_largest:
                num_elements = nums[i] // current_largest + 1
                current_largest = nums[i] // num_elements
            else:
                num_elements = nums[i] // current_largest
            
            total_replacements += num_elements - 1
        
        return total_replacements

File: Minimum_Replacements_to_Sort_the_Array.java
class Solution {
    public long minimumReplacement(int[] nums) {
        int currentLargest = nums[nums.length - 1];
        long totalReplacements = 0;
        
        for (int i = nums.length - 1; i >= 0; i--) {
            if (nums[i] <= currentLargest) {
                currentLargest = nums[i];
                continue;
            }

            int numElements;
            if (nums[i] % currentLargest != 0) {
                numElements = nums[i] / currentLargest + 1;
                currentLargest = nums[i] / numElements;
            } else {
                numElements = nums[i] / currentLargest;
            }
            
            totalReplacements += numElements - 1;
        }
        
        return totalReplacements;        
    }
}
